dias/tracker.py

```diff
@@ -0,0 +1,418 @@
+            for f in files:
+                file_row = FileTable.get(FileTable.filepath == f)
+                file_row.exists = False
+                file_row.save()
+
+    def update_file_table(self):
+        """Insert novel files that are in `self.path` into db, and set exists=False for files that are not at `self.path`, anymore."""
+        files_on_disk = []
+        for ft in ["chimestack", "chimecal", "chimetiming"]:
+            files_on_disk.extend(
+                glob.glob(os.path.join(self.base_path, f"*_{ft}_corr", "*.h5"))
+            )
+
+        files_in_db = [
+            query.filepath
+            for query in (FileTable.select().where(FileTable.exists == True))
+        ]
+
+        self.insert_files([f for f in files_on_disk if f not in files_in_db])
+        self.remove_files([f for f in files_in_db if f not in files_on_disk])
+
+    def new_files(self, update=True):
+        """
+        Return a list of files unprocessed by self.dias_task_name, of its filetype of interest.
+
+        Optionally updates the File Table with the current state of `self.path`.
+
+        Parameters
+        ----------
+        update : Boolean
+            If true, updates the File Table with the current state of `self.path`.
+        """
+        if update:
+            self.update_file_table()
+
+        # Get file type that the analyzer is interested in
+        # assumes analyzer is interested in only one file type
+        file_type_id = (
+            Analyzer.select(Analyzer, FileType)
+            .where(Analyzer.name == self.dias_task_name)
+            .join(FileType)
+        )[0].file_type_id
+
+        # Select files processed by analyzer
+        files_processed = (
+            FileTable.select()
+            .join(Processed)
+            .join(Analyzer)
+            .where(Processed.analyzer_id.name == self.dias_task_name)
+        )
+
+        # SELECT files of analyzer.file_type_id and NOT Processed by Analyzer
+        # TODO Filter by date
+        files_unprocessed = (
+            FileTable.select()
+            .where(FileTable.id.not_in(files_processed))
+            .where(FileTable.file_type_id == file_type_id)
+            .where(FileTable.exists)
+        )
+
+        return [f.filepath for f in files_unprocessed]
+
+    def register_done(self, list_of_files):
+        """
+        Register files in list_of_files as processed by self.dias_task_name.
+
+        Parameters
+        ----------
+        list_of_files : List of Strings
+            List of filepaths to be registered as processed. They need to already exist in the database FileTable.
+        """
+        files_processed = FileTable.select().where(
+            FileTable.filepath.in_(list_of_files)
+        )
+
+        analyzer_id = Analyzer.get(Analyzer.name == self.dias_task_name)
+
+        record = namedtuple("record", "file_id analyzer_id")
+
+        records = [record(f, analyzer_id) for f in files_processed]
+
+        with db.atomic():
```

Comment:
Why do we need this here?

Reply:
I wanted the actual update to the database to be atomic!

Reply:
Why?

Reply:
Because I believed that atomic database updates were good practice. I am not strongly attached to it. Do you have a concern?